sqlx-postgres/src/types/int.rs

```diff
@@ -29,7 +54,11 @@ impl Encode<'_, Postgres> for i8 {
 impl Decode<'_, Postgres> for i8 {
     fn decode(value: PgValueRef<'_>) -> Result<Self, BoxDynError> {
         // note: in the TEXT encoding, a value of "0" here is encoded as an empty string
-        Ok(value.as_bytes()?.get(0).copied().unwrap_or_default() as i8)
+        if (value.format() == PgValueFormat::Text) && (value.as_bytes()?.is_empty()) {
+            return Ok(Default::default());
+        }
+
+        int_decode(value)?.try_into().map_err(Into::into)
```

Comment:
So this behavior isn't the same as what was there before. When the value is in text, it's a single ASCII character. That's why the test is failing.

This is for decoding the `"char"` type: https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/char.c#L58-L60

Even what was there before wasn't technically correct either, for values between `0x80` and `0xFF`.

I think this is what I would do:

```suggestion
        // note: decoding here is for the `"char"` type as Postgres does not have a native 1-byte integer type.
        // https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/char.c#L58-L60
        match value.format() {
            PgValueFormat::Binary => int_decode(value)?.try_into().map_err(Into::into),
            PgValueFormat::Text => {
                let text = value.as_str()?;
                
                // A value of 0 is represented with the empty string.
                if text.is_empty() {
                    return Ok(0);
                }
                
                if text.starts_with('\\') {
                    // For values between 0x80 and 0xFF, it's encoded in octal.
                    return Ok(i8::from_str_radix(text.trim_start_matches('\\'), 8)?);
                }
                
                Ok(text.parse()?)
            }
        }
```

Reply:
Ugh... I really don't like the existence of this impl. If someone tries to use it to decode from an actual integer type it's not going to do the right thing.

I think I'd almost rather get rid of it and add a `PgChar`  type. You don't have to do that here, but I'm probably gonna end up deleting this impl either way because it's just kind of gross.